dimod/discrete/cydiscrete_quadratic_model.pyx

```diff
@@ -291,32 +197,21 @@ cdef class cyDiscreteQuadraticModel:
 
         if array:
             # build a numpy array
-            quadratic = np.zeros((self.num_cases(u), self.num_cases(v)),
+            quadratic = np.empty((self.num_cases(u), self.num_cases(v)),
                                  dtype=self.dtype)
-            quadratic_view = quadratic
-
-            for ci in range(self.case_starts_[u], self.case_starts_[u+1]):
-
-                span = self.bqm_.neighborhood(ci, self.case_starts_[v])
-
-                while (span.first != span.second and deref(span.first).first < self.case_starts_[v+1]):
-                    case_u = ci - self.case_starts_[u]
-                    case_v = deref(span.first).first - self.case_starts_[v]
-                    quadratic_view[case_u, case_v] = deref(span.first).second
-
-                    inc(span.first)
+	    self.dqm_.get_quadratic(u, v, &quadratic[0,0])
 
         else:
             # store in a dict
             quadratic = {}
 
-            for ci in range(self.case_starts_[u], self.case_starts_[u+1]):
+            for ci in range(self.dqm_.case_starts_[u], self.dqm_.case_starts_[u+1]):
```

Comment:
Leaving this part in cython, seems like it is more python specific and will not be needed in the c++ version of the dqm. I assume self.dqm_.case_starts_ .... these are all accessible as usual.